app/main.py

```diff
@@ -1,2 +1,118 @@
+from typing import Any, Optional, Iterator
+
+
 class Dictionary:
-    pass
+    class _Node:
+        def __init__(self, key: Any, value: Any) -> None:
+            self.key = key
+            self.value = value
+            self.next = None
+
+    def __init__(self, initial_capacity: int = 16) -> None:
+        self.capacity: int = initial_capacity
+        self.size: int = 0
+        self.table: list[Optional[self._Node]] = [None] * self.capacity
+
+    def _hash(self, key: Any) -> int:
+        """Generate a hash for the key."""
+        return hash(key) % self.capacity
+
+    def _resize(self) -> None:
+        """Resize the hash table when the load factor is exceeded."""
+        self.capacity *= 2
+        new_table: list[Optional[self._Node]] = [None] * self.capacity
+        for node in self.table:
+            while node:
+                index = self._hash(node.key)
+                new_node = self._Node(node.key, node.value)
+                new_node.next = new_table[index]
+                new_table[index] = new_node
+                node = node.next
+        self.table = new_table
```

Comment:
The `_resize` method doubles the capacity of the hash table and attempts to rehash the nodes. However, the rehashing logic might not correctly place nodes in the new table. Ensure that the `_hash` function is used correctly to compute the new index for each node based on the updated capacity.